decisions/DR013-Project-structure.md

```diff
@@ -0,0 +1,457 @@
+    │   │       └── c
+    │   │           ├── hostApi
+    │   │           └── managerApi
+    │   ├── src
+    │   │   ├── handles
+    │   │   │   ├── hostApi
+    │   │   │   └── managerApi
+    │   │   ├── hostApi
+    │   │   └── managerApi
+    │   └── tests
+    │       └── private
+    │           ├── hostApi
+    │           └── managerApi
+    └── openassetio-python
+        ├── bridge
+        │   ├── include
+        │   │   └── openassetio
+        │   │       └── python
+        │   └── src
+        │       └── python
+        ├── module
+        │   ├── _openassetio
+        │   │   └── src
+        │   │       ├── hostApi
+        │   │       ├── log
+        │   │       └── managerApi
+        │   └── _openassetio_test
+        │       └── src
+        ├── python
+        │   ├── openassetio
+        │   │   ├── _core
+        │   │   ├── hostApi
+        │   │   ├── managerApi
+        │   │   ├── pluginSystem
+        │   │   ├── test
+        │   │   │   └── manager
+        │   │   └── traits
+        │   ├── tests
+        │   │   ├── openassetio
+        │   │   │   ├── _core
+        │   │   │   ├── hostApi
+        │   │   │   │   └── resources
+        │   │   │   ├── managerApi
+        │   │   │   ├── pluginSystem
+        │   │   │   │   └── resources
+        │   │   │   │       ├── pathA
+        │   │   │   │       ├── pathB
+        │   │   │   │       │   └── packagedPlugin
+        │   │   │   │       ├── pathC
+        │   │   │   │       └── symlinkPath
+        │   │   │   ├── test
+        │   │   │   │   └── manager
+        │   │   │   │       └── resources
+        │   │   │   │           └── plugins
+        │   │   │   └── traits
+        │   │   └── openassetio_test
+        │   ├── pyproject.toml
+        │   ├── setup.cfg
+        │   └── setup.py
+        └── shared
+            └── src
+                └── include
+                    └── openassetio
+                        └── private
+                            └── python
+
+```
+
+#### Pros
+
+- Pure Python code is hidden away under nested subdirectories, making
+  it clear at a glance that this is not primarily a Python project.
+- More lightweight repo root directory with less noise.
+- Components can be extracted into their own repo with relative ease.
+- Clearly distinguishes core components that should affect (major)
+  versioning of the project (`/src`) vs. less important snippets used
+  primarily for documentation (`/examples`)
+
+#### Cons
+
+- Duplication of .clang-tidy tests override
+- Test-related helpers (e.g. CMake functions) must move to a common
+  directory, rather than being naturally located next to where they're
+  used, i.e. under a top-level `tests` directory.
+- Duplication of `if (OPENASSETIO_ENABLE_TESTS)` in CMake under each
+  component.
+
+Estimated cost: Medium
+
+
+## Outcome
+
+```mermaid
+flowchart
+    Start --> What{What are\nyou adding?}
+
+    What -->|Docs| WhatDocType{Who is doc for?}
```

Comment:
[fix] Change to simply "Is it docs - put it in `/doc`"

Reply:
Simplified in a17aa50c